delta-lake/src/main/java/org/apache/iceberg/delta/BaseMigrateDeltaLakeTableAction.java

```diff
@@ -0,0 +1,275 @@
+    }
+
+    PartitionSpec.Builder builder = PartitionSpec.builderFor(schema);
+    for (String partitionName : partitionNames) {
+      builder.identity(partitionName);
+    }
+    return builder.build();
+  }
+
+  private void copyFromDeltaLakeToIceberg(Table table, PartitionSpec spec) {
+    Iterator<VersionLog> versionLogIterator =
+        deltaLog.getChanges(
+            0, // retrieve actions starting from the initial version
+            false); // not throw exception when data loss detected
+
+    while (versionLogIterator.hasNext()) {
+      VersionLog versionLog = versionLogIterator.next();
+      List<Action> actions = versionLog.getActions();
+
+      // We first need to iterate through to see what kind of transaction this was. There are 3
+      // cases:
+      // 1. AppendFile - when there are only AddFile instances (an INSERT on the table)
+      // 2. DeleteFiles - when there are only RemoveFile instances (a DELETE where all the records
+      // of file(s) were removed
+      // 3. OverwriteFiles - when there are a mix of AddFile and RemoveFile (a DELETE/UPDATE)
+
+      // Create a map of Delta Lake Action (AddFile, RemoveFile, etc.) --> List<Action>
+      Map<String, List<Action>> deltaLakeActionMap =
+          actions.stream()
+              .filter(action -> action instanceof AddFile || action instanceof RemoveFile)
+              .collect(Collectors.groupingBy(a -> a.getClass().getSimpleName()));
+
+      List<DataFile> filesToAdd = Lists.newArrayList();
+      List<DataFile> filesToRemove = Lists.newArrayList();
+      for (Action action : Iterables.concat(deltaLakeActionMap.values())) {
+        DataFile dataFile = buildDataFileFromAction(action, table, spec);
+        if (action instanceof AddFile) {
+          filesToAdd.add(dataFile);
+        } else if (action instanceof RemoveFile) {
+          filesToRemove.add(dataFile);
+        } else {
+          throw new ValidationException(
+              "The action %s's is unsupported", action.getClass().getSimpleName());
+        }
+      }
+
+      if (filesToAdd.size() > 0 && filesToRemove.size() > 0) {
+        // Overwrite_Files case
+        OverwriteFiles overwriteFiles = table.newOverwrite();
+        filesToAdd.forEach(overwriteFiles::addFile);
+        filesToRemove.forEach(overwriteFiles::deleteFile);
+        overwriteFiles.commit();
+      } else if (filesToAdd.size() > 0) {
+        // Append_Files case
+        AppendFiles appendFiles = table.newAppend();
+        filesToAdd.forEach(appendFiles::appendFile);
+        appendFiles.commit();
+      } else if (filesToRemove.size() > 0) {
+        // Delete_Files case
+        DeleteFiles deleteFiles = table.newDelete();
+        filesToRemove.forEach(deleteFiles::deleteFile);
+        deleteFiles.commit();
+      }
+    }
+  }
+
+  private DataFile buildDataFileFromAction(Action action, Table table, PartitionSpec spec) {
+    String path;
+    Optional<Long> nullableSize;
```

Comment:
looks like we don't even need this. Because in `FileMetricsReader.getMetricsForFile`, we will always create an input file no matter what, so we can always derive the correct file size instead of relying on the information in the action.